### PR TITLE
Question button which opens a question form within a wiki #1453

### PR DESCRIPTION
**Pull request by aspriya**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

**Comment by ananyo2012** [comment]
@aspriya Can you try to write a jasmine test for this?

**Comment by ananyo2012** [comment]
@aspriya Are you facing any problem in merging/rebasing the upstream changes? I see multiple merge commits again. Actually a lot of them. This isn't a big issue since we can always squash them while merging. But it is good to know how to do it in a proper way.

**Comment by jywarren** [comment]
Wow, if it opens a questions form in a wiki, that may be one step further
than we needed to go in the initial version. Remember to break things into
separate parts so you can complete two smaller projects more iteratively
than one larger project. Can you share screenshots of the sequence that
results?

But in any case if you've gone ahead to do this, that's fine -- it's great
that you've gotten this far. Tell us when you're prepared to have it tested
out!

I'm not entirely sure a Jasmine test will work, will it? Could we add one
which sends a click event to the button, then checks if the form is
visible?

Thanks!

**Comment by jywarren** [comment]
Ah, no, looking at the changed files, i think this just adds the extra button and opens the form in a new window. That's just right.

Is this ready to be merged then? I think it's such a simple feature, and using just the `inline-markdown-editor` API, that we don't need a test for it. If it had tighter integration with the backend, such as creating a form inline, we might think of tests. But here I think it's OK.

It'd be great to be able to show this to people on the live site and get feedback, and satisfying to see it published! Tell us when it's ready for merging!

**Comment by jywarren** [comment]
And, once it is, could you post a research note on PublicLab.org introducing it, and showing step by step how people can use it, with a screenshot? That would be a great way to get peoples' feedback. I can tell you when it gets published to production once you give us the sign that it's ready to be merged.

**Comment by aspriya** [comment]
@jywarren, So with my latest push, now a question form opens in a new tab and that opened question forms has 3 tags like below.
`question:<title-of-the-wiki>`    eg: `question:nano-data-logger`
`<title-of-the-wiki-section-form-id>`   eg: `nano-data-logger-section-form-3935`
`a-wiki-question`

By these tags we can distuingish questions as, to what wiki is this question is relating and also to what section of that wiki.

**Comment by aspriya** [comment]
If there are questions being asked about a wiki page, how about if we add a power tag to be displayed automatically in that wiki page?. And more over how about if we do this in section wise as now there is a way to track the quesitons relating a certain section of the wiki via the tag `<title-of-the-wiki-section-form-id>`

**Comment by aspriya** [comment]
@ananyo2012, Here's what I normally do, 
I get a pull from `https://github.com/publiclab/plots2.git`. and after that I create a new feature branch. And keep working on that. Then when I'm done, I checkout againg to master branch and do a pull again from `https://github.com/publiclab/plots2.git`. After that I checkout to my feature branch again and do a `git rebase master` And resolve merge conflicts if any. 

After that I run `rake test`. If all are ok, I push the code to my remote repo which is `git@github.com:aspriya/plots2.git` by the command `git push origin <my_feature_branch_name>` And from there I make a pull request to master branch of plots2. That's my workflow. 

Is there something I do wrong here? I also see those merge commits. how can I remove those? Some times may because of above mentioned workflow and I'm doing something wrong. pls help me.

**Comment by aspriya** [comment]
@jywarren, [Here](https://publiclab.org/notes/Ashan/06-14-2017/introducing-question-button-in-wiki-pages?_=1497434019) I posted a new research note about this feature. So if it's ok, I think this is ok to be merged.

File: app/assets/javascripts/wikis.js [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
       replaceUrl: '/wiki/replace/' + node_id,
       selector: '#content-raw-markdown',
       wysiwyg: true,
       preProcessor: preProcessMarkdown,
-      postProcessor: postProcessContent
+      postProcessor: postProcessContent,
+      extraButtons: {"fa-question": questionForm}
     });
     $('#content').hide();
+    console.log(wiki_title);
```
Comment (jywarren) [inline app/assets/javascripts/wikis.js]: Can you remove this line? Thanks!

**Comment by jywarren** [comment]
Actually since your post is already up, I made the changes (they were very small) and will merge now!

**Comment by jywarren** [comment]
https://publiclab.org/notes/Ashan/06-14-2017/introducing-question-button-in-wiki-pages

**Comment by ananyo2012** [comment]
@aspriya The workflow seems fine. I checked out your master branch at https://github.com/publiclab/plots2/commits/master. It contains some 4 commits at the top which shouldn't be there. I guess that is causing the problems. Can you try removing them  using `git reset --hard HEAD~4` ?

**Pull request merged**